src/cosmicqc/scdataframe.py

```diff
@@ -0,0 +1,164 @@
+"""
+Defines a SCDataFrame class for use in coSMicQC.
+"""
+
+import pathlib
+from typing import Any, Dict, TypeVar, Union
+
+import pandas as pd
+
+# provide backwards compatibility for Self type in earlier Python versions.
+# see: https://peps.python.org/pep-0484/#annotating-instance-and-class-methods
+SCDataFrame_type = TypeVar("SCDataFrame_type", bound="SCDataFrame")
+
+
+class SCDataFrame:
+    """
+    A class to handle and load different types of data files into a pandas DataFrame.
+
+    This class can initialize with either a pandas DataFrame or a file path (CSV, TSV,
+    TXT, or Parquet). When initialized with a file path, it reads the data into a
+    pandas DataFrame.
+
+    Attributes:
+        data_source (str):
+            A string indicating the data source, either 'pd.DataFrame'
+            or the file path.
+        data (pd.DataFrame):
+            The loaded data in a pandas DataFrame.
+
+    Methods:
+        __call__():
+            Returns the underlying pandas DataFrame.
+        __repr__():
+            Returns a representational string of the underlying pandas DataFrame.
+        __getattr__():
+            Returns the underlying attributes of the pandas DataFrame.
+        __getitem__():
+            Returns slice of data from pandas DataFrame.
+    """
+
+    def __init__(
+        self: SCDataFrame_type,
+        data: Union[SCDataFrame_type, pd.DataFrame, str, pathlib.Path],
+        **kwargs: Dict[str, Any],
+    ) -> None:
+        """
+        Initializes the SCDataFrame with either a DataFrame or a file path.
+
+        Args:
+            data (Union[pd.DataFrame, str]):
+                The data source, either a pandas DataFrame or a file path.
+            **kwargs:
+                Additional keyword arguments to pass to the pandas read functions.
+        """
+
+        if isinstance(data, SCDataFrame):
+            # if data is an instance of SCDataFrame, use its data_source and data
+            self.data_source = data.data_source
+            self.data = data.data
+
+        elif isinstance(data, pd.Series):
+            # if data is a pd.Series, remember this within the data_source attr
+            self.data_source = "pandas.Series"
+            # also cast the series to a dataframe
+            self.data = pd.DataFrame(data)
+
+        elif isinstance(data, pd.DataFrame):
+            # if data is a pd.DataFrame, remember this within the data_source attr
+            self.data_source = "pandas.DataFrame"
+            self.data = data
+
+        elif isinstance(data, (pathlib.Path, str)):
+            # if the data is a string, remember the original source
```

Comment:
```suggestion
            # if the data is a string or a pathlib path, remember the original source
```
Would this be a correct update to the comment?

Reply:
Nice catch, yes that's accurate!